tests/runner.py

```diff
@@ -572,13 +579,12 @@ def verify_es5(self, filename):
   def build(self, filename, libraries=[], includes=[], force_c=False,
             post_build=None, js_outfile=True, emcc_args=[]):
     suffix = '.js' if js_outfile else '.wasm'
-    if shared.suffix(filename) in ('.cc', '.cxx', '.cpp') and not force_c:
-      compiler = [EMXX]
-    else:
+    compiler = [compiler_for(filename, force_c)]
+    if compiler[0] == EMCC:
       # TODO(https://github.com/emscripten-core/emscripten/issues/11121)
       # We link with C++ stdlibs, even when linking with emcc for historical reasons.  We can remove
       # this if this issues is fixed.
-      compiler = [EMCC, '-nostdlib++']
+      compiler.append('-nostdlib++')
```

Comment:
do we still need this? (the issue is still open so I guess so?)

Reply:
Yes, unless we switch the default, which I hope to soon, this is still useful to catch accidental deps on libc++.  Its especially useful for MAIN_MODULE=1 programs are written in C (without this option such programs get a copy of libc++ even though they don't need it!)